pkg/config/config.go

```diff
@@ -290,15 +295,71 @@ type IngressRecommendations struct {
 	TLSSecretValid *bool `yaml:"tlsSecretValid,omitempty"`
 }
 
+// providesPlugins defines behavior for providing Plugins
+type providesPlugins interface {
+	GetPlugins() Plugins
+}
+
 // Plugins contains plugins configuration parameters defined in groups.
 type Plugins map[string]Plugin
 
 // Plugin contains plugin specific configuration.
 type Plugin struct {
 	Enabled bool
 	Config  any
+	Context PluginContext
+}
+
+// PluginContext defines the context for given plugin.
+type PluginContext struct {
+	// Kubeconfig defines configuration options for plugins kubeconfig.
+	Kubeconfig KubeconfigContext
+	// RBAC defines the RBAC rules for given plugin.
+	RBAC PolicyRule
+}
+
+type KubeconfigContext struct {
+	// DefaultNamespace is the default namespace for this kubeconfig.
+	DefaultNamespace string
+}
+
+// PolicyRule is the RBAC rule.
+type PolicyRule struct {
+	// User is the policy subject for user.
+	User PolicySubject
+	// Group is the policy subject for group.
+	Group PolicySubject
 }
 
+type PolicySubject struct {
```

Comment:
Please add a comment and make sure other exported types/function/methods also have it 👍